src/ripple/app/tests/Path_test.cpp

```diff
@@ -180,12 +190,117 @@ find_paths(jtx::Env& env,
             std::move(sa), std::move(da));
 }
 
+Json::Value
+rpf(jtx::Account const& src, jtx::Account const& dst, std::uint32_t num_src)
+{
+    Json::Value jv = Json::objectValue;
+    jv[jss::command] = "ripple_path_find";
+    jv[jss::source_account] = toBase58(src);
+
+    if (num_src > 0)
+    {
+        auto& sc = (jv[jss::source_currencies] = Json::arrayValue);
+        Json::Value j = Json::objectValue;
+        while (num_src--)
+        {
+            j[jss::currency] = std::to_string(num_src + 100);
+            sc.append(j);
+        }
+    }
+
+    auto const d = toBase58(dst);
+    jv[jss::destination_account] = d;
+
+    Json::Value& j = (jv[jss::destination_amount] = Json::objectValue);
+    j[jss::currency] = "USD";
+    j[jss::value] = "0.01";
+    j[jss::issuer] = d;
+
+    return jv;
+}
+
 //------------------------------------------------------------------------------
 
 class Path_test : public beast::unit_test::suite
 {
 public:
     void
+    source_currencies_limit()
+    {
+        /* TODO MPORTILLA Commented out until Vinnie's mao branch is merged
+        using namespace std::chrono_literals;
+        using namespace jtx;
+        testcase("source currency limits");
+        Env env(*this);
+        auto const gw = Account("gateway");
+        env.fund(XRP(10000), "alice", "bob", gw);
+        env.trust(gw["USD"](100), "alice", "bob");
+        env.close(env.now());
+
+        auto& app = env.app();
+        Resource::Charge loadType = Resource::feeReferenceRPC;
+        RPC::Context context {beast::Journal(), {}, app, loadType, app.getOPs(),
+            app.getLedgerMaster(), Role::USER, {}};
+        std::mutex m;
+        std::condition_variable cv;
+        Json::Value result;
+
+        // Test RPC::Tuning::max_src_cur source currencies.
+        app.getJobQueue().postCoro(jtCLIENT, "RPC-Client",
+            [&](std::shared_ptr<JobCoro> jc)
+            {
+                context.params = rpf(Account("alice"), Account("bob"),
+                    RPC::Tuning::max_src_cur);
+                context.jobCoro = jc;
+                RPC::doCommand(context, result);
+                cv.notify_one();
+            });
+
+        {
+            std::unique_lock<std::mutex> lk(m);
+            cv.wait_for(lk, 2s);
+            expect(! result.isMember(jss::error));
+        }
+
+        // Test more than RPC::Tuning::max_src_cur source currencies.
+        app.getJobQueue().postCoro(jtCLIENT, "RPC-Client",
+            [&](std::shared_ptr<JobCoro> jc)
+            {
+                context.params = rpf(Account("alice"), Account("bob"),
+                    RPC::Tuning::max_src_cur + 1);
```

Comment:
It might be nice to have another test case that goes right up to the limit (ie. without the `+ 1`) and succeeds to confirm there are no fencepost-type errors.